packages/wallet-sdk/src/connection/WalletLinkConnection.ts

```diff
@@ -66,7 +57,25 @@ export class WalletLinkConnection {
   private nextReqId = IntNumber(1);
   private connectedSubject = new BehaviorSubject(false);
   private linkedSubject = new BehaviorSubject(false);
-  private sessionConfigSubject = new ReplaySubject<SessionConfig>(1);
+
+  private sessionConfigListner?: (_: SessionConfig) => void;
+  setSessionConfigListner(listener: (_: SessionConfig) => void): void {
+    this.sessionConfigListner = listener;
+  }
+
+  private linkedListner?: (_: boolean) => void;
+  setLinkedListner(listener: (_: boolean) => void): void {
+    this.linkedListner = listener;
+  }
+
+  private connectedListner?: (_: boolean) => void;
+  setConnectedListner(listener: (_: boolean) => void): void {
+    this.connectedListner = listener;
+  }
+
+  setIncomingEventListner(listener: (_: ServerMessageEvent) => void): void {
+    this.subscribeIncomingEvent(listener);
```

Comment:
instead of exposing `incomingEvent$` observable,
now it has `setIncomingEventListner`, and handles subscription within itself when new listener is set.